roofit/roofitcore/test/testRooFuncWrapper.cxx

```diff
@@ -188,111 +188,11 @@ TEST(RooFuncWrapper, NllWithObservables)
-      return "-1e30";
-   return std::to_string(val);
-}
-
-std::string integralCode(RooRealIntegral const &integral)
-{
-   std::stringstream ss;
-   // TODO: assert also that numIntCatVars() and numIntRealVars() are empty
-
-   RooArgSet anaIntVars{integral.anaIntVars()};
-   RooArgSet empty;
-   auto range = integral.intRange();
-   const int code = integral.integrand().getAnalyticalIntegralWN(anaIntVars, empty, nullptr, range);
-   if (code == 1) {
-      auto gauss = dynamic_cast<RooGaussian const *>(&integral.integrand());
-      RooRealVar const &x = static_cast<RooRealVar const &>(gauss->getX());
-      RooRealVar const &mean = static_cast<RooRealVar const &>(gauss->getMean());
-      ss << "RooFit::Detail::AnalyticalIntegrals::gaussianIntegral(" << valToString(x.getMin(range)) << ", "
-         << valToString(x.getMax(range)) << ", " << valName(mean) << ", " << valName(gauss->getSigma()) << ")";
-   }
-   return ss.str();
-}
-
-std::string gaussianCode(RooGaussian const &gauss)
-{
-   std::stringstream ss;
-
-   ss << "GaussianEval(" << valName(gauss.getX()) << ", " << valName(gauss.getMean()) << ", "
-      << valName(gauss.getSigma()) << ")";
-
-   return ss.str();
-}
-
-std::string normalizedPdfCode(RooAbsArg const &pdf)
-{
-   std::stringstream ss;
-
-   ss << valName(*pdf.servers()[0]) << " / " << valName(*pdf.servers()[1]);
-
-   return ss.str();
-}
-
-std::string generateCode(RooAbsReal const &func, RooArgSet const &variables)
-{
-
-   RooArgSet nodes;
-   RooHelpers::getSortedComputationGraph(func, nodes);
-
-   std::stringstream ss;
-
-   for (RooAbsArg *node : nodes) {
-      auto var = dynamic_cast<RooRealVar *>(node);
-      int idx = variables.index(node);
-      if (var && idx >= 0) {
-         ss << "const double " << valName(*var) << " = params[" << idx << "];\n";
-      } else if (var) {
-         ss << "const double " << valName(*var) << " = " << var->getVal() << ";\n";
-      } else if (auto gauss = dynamic_cast<RooGaussian *>(node)) {
-         ss << "const double " << valName(*gauss) << " = " << gaussianCode(*gauss) << ";\n";
-      } else if (auto integral = dynamic_cast<RooRealIntegral *>(node)) {
-         ss << "const double " << valName(*integral) << " = " << integralCode(*integral) << ";\n";
-      } else if (node == &func) {
-         ss << "const double " << valName(*node) << " = " << normalizedPdfCode(*node) << ";\n";
-      }
-   }
-   ss << "return " << valName(func) << ";\n";
-
-   std::string out = ss.str();
-
-   return out;
-}
-
-} // namespace
-
 TEST(RooFuncWrapper, GaussianNormalized)
 {
    using namespace RooFit;
 
-   // clang-format off
-   gInterpreter->Declare(
-   "double GaussianEval(double x, double mean, double sigma)"
-   "{"
-   "   const double arg = x - mean;"
-   "   return std::exp(-0.5 * arg * arg / (sigma * sigma));"
-   "}"
-   );
-   // clang-format on
-
-   RooRealVar x("x", "x", 0, -10, 10);
+   RooRealVar x("x", "x", 0, -10, std::numeric_limits<double>::infinity());
```

Comment:
```c++
   RooRealVar x("x", "x", 0, -10, std::numeric_limits<double>::infinity());

   RooRealVar mu("mu", "mu", 0, -10, 10);
   RooRealVar shift("shift", "shift", 1.0, -10, 10);
   RooAddition muShifted("mu_shifted", "mu_shifted", mu, shift);

   RooRealVar sigma("sigma", "sigma", 2.0, 0.01, 10);
   RooConstVar scale("scale", "scale", 1.5);
   RooProduct sigmaScaled("sigma_scaled", "sigma_scaled", sigma, scale);

   RooGaussian gauss{"gauss", "gauss", x, muShifted, sigmaScaled};
```
I suggest a slightly more complicated testcase with addition and product to test everything you have implemented so far.